modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/redownload/AlignmentCheck.scala

```diff
@@ -0,0 +1,89 @@
+
+import scala.collection.immutable.SortedSet
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+import org.tessellation.ext.collection.IterableUtils.pickMajority
+import org.tessellation.schema.peer.{L0Peer, Peer, PeerInfo}
+import org.tessellation.schema.{GlobalIncrementalSnapshot, GlobalSnapshotInfo}
+import org.tessellation.sdk.domain.Daemon
+import org.tessellation.sdk.domain.cluster.storage.{ClusterStorage, L0ClusterStorage}
+import org.tessellation.sdk.domain.seedlist.SeedlistEntry
+import org.tessellation.sdk.domain.snapshot.storage.LastSnapshotStorage
+import org.tessellation.sdk.http.p2p.clients.L0GlobalSnapshotClient
+
+trait AlignmentCheck[F[_]] {
+  def check(): F[Unit]
+}
+
+object AlignmentCheck {
+  private val maxConcurrentPeerInquiries = 10
+  def make[F[_]: Async: Temporal](
+    priorityPeers: NonEmptySet[SeedlistEntry],
+    isAlignedR: Ref[F, Boolean],
+    clusterStorage: ClusterStorage[F],
+    l0ClusterStorage: L0ClusterStorage[F],
+    lastSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
+    l0GlobalSnapshot: L0GlobalSnapshotClient[F],
+    waitForPeers: FiniteDuration = 1.minute
+  ): AlignmentCheck[F] = () => {
+    // TODO Are these the correct peers to use to determine if we're aligned?
+    //  At first I thought to use the priority seed list but I'm not sure that's correct
+    def getPeers: F[List[Peer]] =
+      priorityPeers.toNonEmptyList
+        .traverse(se => clusterStorage.getPeer(se.peerId))
+        .map(_.toList.flatten)
+
+    def toL0Peers(peers: List[Peer]): List[L0Peer] =
+      peers.map(p => L0Peer.fromPeerInfo(PeerInfo.fromPeer(p)))
+
+    for {
+      (ordinal, hash) <- lastSnapshotStorage.get
+        .flatMap(MonadThrow[F].fromOption(_, new Exception("TODO raise error or exit early?")))
+        .map(hs => (hs.ordinal, hs.hash))
+
+      // TODO add comment explaining why we're sleeping
+      _ <- Temporal[F].sleep(waitForPeers)
+
+      // TODO filter on Responsive?
+      // TODO what to do if empty?
+      //   do nothing and try again next iteration?
+      //   will that ever happen?
+      peers <- getPeers
+
+      majorityHash <- peers
+        .parTraverseN(maxConcurrentPeerInquiries)(p => l0GlobalSnapshot.getHash(ordinal).run(p))
+        .map(_.flatten)
+//        .flatMap(hs => MonadThrow[F].fromOption(hs.toNel, new Exception("No peers have our ordinal")))
+        .map(pickMajority)
+
+      // TODO if we are not aligned, can we pick any peer from seedlist and use that as new global0Peer ?
+      //  We can store it in the Ref, with the initial value being the either cfg or method globalL0Peer
+      isAligned = majorityHash.fold(false)(hash === _)
+
+      // TODO alternatively, we can reset our l0 peers using priority seed list?
+      //   if so, is this the place to do it?
+      maybeL0Peers = SortedSet.from(toL0Peers(peers)).toNes
+      _ <- maybeL0Peers.traverse_(l0ClusterStorage.setPeers).unlessA(isAligned)
```

Comment:
I don't think you should modify it here, we already have a process that modifies L0 peers based on the last processed majority snapshot. If you don't let the L0Service to download a non majority one, the peers will be set correctly based on the signers of a majority snapshot.

Reply:
Agreed, regarding discussion from meeting: if we do not modify the peers but instead accept only from snapshot we won't need to adjust these but rather make the assumption that those peers will be maintained in the list originating from the user supplied variable and adjusted based on alignment with correct latest snapshot.

Reply:
@ryle-goehausen @TheMMaciek @marcinwadon 

This is thread is outdated -- the code was changed a long time ago. This is one of the reasons why I asked we be more diligent about closing off conversations that are no longer relevant.

Reply:
If you consider a conversation outdated or resolved you can resolve it, it doesn't have to be the reviewer, it's actually better I guess to have the PR owner resolve conversations. Marking that the issue was checked/understood and either addressed or is not relevant anymore.